proposals/0417-task-executor-preference.md

```diff
@@ -679,19 +736,19 @@ Thanks to the equivalence between `SomeGlobalActor.shared` instance and `@SomeGl
 @MainActor 
 var example: Int = 0
 
-Task(on: MainActor.shared) { 
+Task(executorPreference: MainActor.shared) { 
    example = 12 // not crossing actor-boundary
 }
 ```
 
-It is more efficient to write `Task(on: MainActor.shared) {}` than it is to `Task { @MainActor in }` because the latter will first launch the task on the inferred context (either enclosing actor, or global concurrent executor), and then hop to the main actor. The `on MainActor` spelling allows Swift to immediately enqueue on the actor itself.
+It is more efficient to write `Task(executorPreference: MainActor.shared) {}` than it is to `Task { @MainActor in }` because the latter will first launch the task on the inferred context (either enclosing actor, or global concurrent executor), and then hop to the main actor. The `on MainActor` spelling allows Swift to immediately enqueue on the actor itself.
```

Comment:
```suggestion
It is more efficient to write `Task(executorPreference: MainActor.shared) {}` than it is to `Task { @MainActor in }` because the latter will first launch the task on the inferred context (either enclosing actor, or global concurrent executor), and then hop to the main actor. The `executorPreference: MainActor` spelling allows Swift to immediately enqueue on the actor itself.
```